app/components/Sidebar/components/CollectionLink.js

```diff
@@ -74,49 +78,103 @@ function CollectionLink({
     }),
   });
 
+  //Drop to reorder Collection
+  const [{ isCollectionDropping }, dropToReorderCollection] = useDrop({
+    accept: "collection",
+    drop: async (item, monitor) => {
+      collections.move(
+        item.id,
+        fractionalIndex(collection.index, belowCollectionIndex)
+      );
+    },
+    canDrop: (item, monitor) => {
+      return (
+        collection.id !== item.id &&
+        (!belowCollection || item.id !== belowCollection.id)
+      );
+    },
+    collect: (monitor) => ({
+      isCollectionDropping: monitor.isOver(),
+    }),
+  });
+
+  // Drag to reorder Collection
+  const [{ isCollectionDragging }, dragToReorderCollection] = useDrag({
+    item: {
+      type: "collection",
+      activeCollectionId: ui.activeCollectionId,
+      id: collection.id,
+    },
+    collect: (monitor) => ({
+      isCollectionDragging: monitor.isDragging(),
+    }),
+    canDrag: (monitor) => {
+      return can.move;
+    },
+    begin: (monitor) => {
+      ui.activeCollectionId = "";
+    },
+    end: (monitor) => {
+      ui.activeCollectionId = monitor.activeCollectionId;
+    },
```

Comment:
What I think you're attempting here is to temporarily hide the collection contents when a drag is active.
- modifying central state like this is a bit of a roundabout way to accomplish this goal
- instead, move the calculation for `expanded` https://github.com/outline/outline/pull/1870/files#diff-c6ce8fd8c9fd8e2796d41937514439cadff8e2d64b3306caf77a4e4fb07d9901R47 below here and simply change its definition to also include `!isCollectionDragging`